Content.Server/Atmos/TileAtmosphere.cs

```diff
@@ -0,0 +1,718 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Server.Interfaces.Atmos;
+using Content.Shared.Atmos;
+using Content.Shared.GameObjects.EntitySystems;
+using NFluidsynth;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Components;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Map;
+using Robust.Shared.Interfaces.Random;
+using Robust.Shared.IoC;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+using Robust.Shared.Random;
+using Robust.Shared.Utility;
+using Logger = Robust.Shared.Log.Logger;
+using Math = CannyFastMath.Math;
+using MathF = CannyFastMath.MathF;
+
+namespace Content.Server.Atmos
+{
+    public class TileAtmosphere
+    {
+        private static long _eqQueueCycleCtr = 0;
+        private int _archivedCycle = 0;
+        private int _currentCycle = 0;
+
+        public TileAtmosphere PressureSpecificTarget { get; set; } = null;
+        public float PressureDifference { get; set; } = 0;
+        public int AtmosCooldown { get; set; } = 0;
+        public bool Excited { get; set; } = false;
+        private IGridAtmosphereManager _gridAtmosphereManager;
+
+        private readonly Dictionary<Direction, TileAtmosphere> _adjacentTiles = new Dictionary<Direction, TileAtmosphere>();
+
+        private readonly TileAtmosInfo _tileAtmosInfo = new TileAtmosInfo()!;
+        private Direction _pressureDirection;
+
+        public MapId MapIndex { get; }
```

Comment:
Map index can change if the grid gets moved to a different map (think mining shuttle in SS13).

It also isn't used so you're best off getting rid of it.